### ISSUE TITLE
Realtek bluetooth 5 dongle fails from time to time

### ISSUE THREAD

**Issue by clau-bucur** [home-assistant/operating-system#1273]
**Hardware Environment**

- [ ] Raspberry Pi [1/2/3/4]
- [ ] ODROID [C2/C4/N2(+)/XU4]
- [ ] ASUS Tinker [S]
- [ ] Generic x86-64 (like Intel NUC)
- [x ] OVA (Open Virtualization Applicance, on Intel NUC or any other hardware, please add the Hypervisor you are using)

**Home Assistant OS release:**
- [ ] Fresh installation of release x.y
- [x] Updated from version 5.12
- Additional information (if accessible):

## System Health

version | core-2021.3.3
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
virtualenv | false
python_version | 3.8.7
os_name | Linux
os_version | 5.4.99
arch | x86_64
timezone | Europe/Bucharest

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
Github API Calls Remaining | 4774
Installed Version | 1.11.3
Stage | running
Available Repositories | 833
Installed Repositories | 26

</details>

<details><summary>AccuWeather</summary>

can_reach_server | ok
-- | --
remaining_requests | 11

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | true
-- | --
subscription_expiration | 12 April 2021, 03:00
relayer_connected | true
remote_enabled | false
remote_connected | false
alexa_enabled | false
google_enabled | true
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 5.12
-- | --
update_channel | stable
supervisor_version | supervisor-2021.03.4
docker_version | 19.03.13
disk_total | 19.0 GB
disk_used | 12.0 GB
healthy | true
supported | true
board | ova
supervisor_api | ok
version_api | ok
installed_addons | Home Assistant Google Drive Backup (0.103.1), File editor (5.2.0), TasmoAdmin (0.14.0), Samba share (9.3.1), MariaDB (2.2.2), Nginx Proxy Manager (0.10.0), NetDaemon (21.08.1), Grafana (6.1.3), InfluxDB (4.0.3), SSH & Web Terminal (8.0.3), TasmoBackup (1.04.10), Portainer (1.4.0), Paradox Alarm Interface (Development version) (dev-latest), Mosquitto broker (5.1.1)

</details>

<details><summary>Lovelace</summary>

dashboards | 2
-- | --
resources | 25
views | 25
mode | yaml

</details>
**Supervisor logs:**

```
21-03-15 07:56:21 INFO (SyncWorker_3) [supervisor.docker.interface] Attaching to homeassistant/amd64-addon-mosquitto with version 5.1.1
21-03-15 07:56:21 INFO (SyncWorker_1) [supervisor.docker.interface] Attaching to homeassistant/amd64-addon-mariadb with version 2.2.2
21-03-15 07:56:21 INFO (SyncWorker_2) [supervisor.docker.interface] Attaching to danmed/tasmobackupv1 with version 1.04.10
21-03-15 07:56:21 INFO (SyncWorker_6) [supervisor.docker.interface] Attaching to ghcr.io/hassio-addons/nginxproxymanager/amd64 with version 0.10.0
21-03-15 07:56:21 INFO (MainThread) [supervisor.snapshots] Found 7 snapshot files
21-03-15 07:56:21 INFO (MainThread) [supervisor.discovery] Loaded 1 messages
21-03-15 07:56:21 INFO (MainThread) [supervisor.ingress] Loaded 0 ingress sessions
21-03-15 07:56:21 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.SETUP
21-03-15 07:56:21 INFO (MainThread) [supervisor.resolution.check] System checks complete
21-03-15 07:56:21 INFO (MainThread) [supervisor.resolution.evaluate] Starting system evaluation with state CoreState.SETUP
21-03-15 07:56:21 INFO (MainThread) [supervisor.resolution.evaluate] System evaluation complete
21-03-15 07:56:21 INFO (MainThread) [__main__] Running Supervisor
21-03-15 07:56:21 INFO (MainThread) [supervisor.hassos] Rauc: A - marked slot kernel.0 as good
21-03-15 07:56:21 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.INITIALIZE' starting 0 add-ons
21-03-15 07:56:21 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.SYSTEM' starting 2 add-ons
21-03-15 07:56:21 INFO (SyncWorker_7) [supervisor.docker.interface] Cleaning addon_core_mariadb application
21-03-15 07:56:22 INFO (SyncWorker_7) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-mariadb with version 2.2.2
21-03-15 07:56:22 INFO (SyncWorker_6) [supervisor.docker.interface] Cleaning addon_core_mosquitto application
21-03-15 07:56:23 INFO (SyncWorker_6) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-mosquitto with version 5.1.1
21-03-15 07:56:24 INFO (MainThread) [supervisor.services.modules.mqtt] Set core_mosquitto as service provider for mqtt
21-03-15 07:56:24 INFO (MainThread) [supervisor.services.modules.mysql] Set core_mariadb as service provider for MySQL
21-03-15 07:56:24 INFO (MainThread) [supervisor.auth] Auth request from 'core_mosquitto' for 'gmemqtt'
21-03-15 07:56:24 INFO (MainThread) [supervisor.auth] Home Assistant not running, checking cache
21-03-15 07:56:28 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.SERVICES' starting 5 add-ons
21-03-15 07:56:28 INFO (SyncWorker_1) [supervisor.docker.interface] Cleaning addon_core_samba application
21-03-15 07:56:28 INFO (SyncWorker_1) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-samba with version 9.3.1
21-03-15 07:56:28 INFO (SyncWorker_5) [supervisor.docker.interface] Cleaning addon_a0d7b954_grafana application
21-03-15 07:56:29 INFO (SyncWorker_5) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/grafana/amd64 with version 6.1.3
21-03-15 07:56:29 INFO (SyncWorker_7) [supervisor.docker.interface] Cleaning addon_a0d7b954_influxdb application
21-03-15 07:56:29 INFO (SyncWorker_7) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/influxdb/amd64 with version 4.0.3
21-03-15 07:56:29 WARNING (SyncWorker_1) [supervisor.docker.addon] SSH & Web Terminal running with disabled protected mode!
21-03-15 07:56:29 INFO (SyncWorker_1) [supervisor.docker.interface] Cleaning addon_a0d7b954_ssh application
21-03-15 07:56:30 INFO (SyncWorker_1) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/ssh/amd64 with version 8.0.3
21-03-15 07:56:30 WARNING (SyncWorker_7) [supervisor.docker.addon] Portainer running with disabled protected mode!
21-03-15 07:56:30 INFO (SyncWorker_7) [supervisor.docker.interface] Cleaning addon_a0d7b954_portainer application
21-03-15 07:56:30 INFO (SyncWorker_7) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/portainer/amd64 with version 1.4.0
21-03-15 07:56:30 INFO (MainThread) [supervisor.api.security] /dns/info access from a0d7b954_influxdb
21-03-15 07:56:35 INFO (MainThread) [supervisor.core] Start Home Assistant Core
21-03-15 07:56:35 INFO (SyncWorker_5) [supervisor.docker.interface] Starting homeassistant
21-03-15 07:56:35 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
21-03-15 07:56:43 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
21-03-15 07:57:30 INFO (MainThread) [supervisor.homeassistant.core] Detect a running Home Assistant instance
21-03-15 07:57:30 INFO (MainThread) [supervisor.addons] Phase 'AddonStartup.APPLICATION' starting 5 add-ons
21-03-15 07:57:30 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password
21-03-15 07:57:30 INFO (SyncWorker_6) [supervisor.docker.interface] Cleaning addon_cebe7a76_hassio_google_drive_backup application
21-03-15 07:57:30 INFO (SyncWorker_6) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password
21-03-15 07:57:31 INFO (SyncWorker_6) [supervisor.docker.addon] Starting Docker add-on sabeechen/hassio-google-drive-backup-amd64 with version 0.103.1
21-03-15 07:57:31 INFO (SyncWorker_3) [supervisor.docker.interface] Cleaning addon_core_configurator application
21-03-15 07:57:31 INFO (SyncWorker_3) [supervisor.docker.addon] Starting Docker add-on homeassistant/amd64-addon-configurator with version 5.2.0
21-03-15 07:57:31 INFO (SyncWorker_4) [supervisor.docker.interface] Cleaning addon_a0d7b954_nginxproxymanager application
21-03-15 07:57:32 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /core/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /supervisor/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /snapshots access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /snapshots/65461f7a/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /snapshots/2cd2e9da/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /snapshots/69cb4334/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /snapshots/0df44dca/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /snapshots/7c9848a4/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /snapshots/934db106/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:32 INFO (MainThread) [supervisor.api.security] /snapshots/eb200b0b/info access from cebe7a76_hassio_google_drive_backup
21-03-15 07:57:33 INFO (SyncWorker_4) [supervisor.docker.addon] Starting Docker add-on ghcr.io/hassio-addons/nginxproxymanager/amd64 with version 0.10.0
21-03-15 07:57:33 INFO (SyncWorker_2) [supervisor.docker.interface] Cleaning addon_c6a2317c_netdaemon application
21-03-15 07:57:33 INFO (SyncWorker_2) [supervisor.docker.addon] Starting Docker add-on netdaemon/netdaemon with version None
21-03-15 07:57:33 INFO (SyncWorker_7) [supervisor.docker.interface] Cleaning addon_e92a3241_paradox_alarm_interface_dev application
21-03-15 07:57:34 INFO (SyncWorker_7) [supervisor.docker.addon] Starting Docker add-on paradoxalarminterface/pai with version None
21-03-15 07:57:34 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request initialize
21-03-15 07:57:34 INFO (MainThread) [supervisor.api.security] /dns/info access from a0d7b954_nginxproxymanager
21-03-15 07:57:34 INFO (MainThread) [supervisor.api.proxy] WebSocket access from c6a2317c_netdaemon
21-03-15 07:57:34 INFO (MainThread) [supervisor.api.proxy] Home Assistant WebSocket API request running
21-03-15 07:57:39 INFO (MainThread) [supervisor.misc.tasks] All core tasks are scheduled
21-03-15 07:57:39 INFO (MainThread) [supervisor.core] Supervisor is up and running
21-03-15 07:57:39 INFO (MainThread) [supervisor.host.info] Updating local host information
21-03-15 07:57:39 INFO (MainThread) [supervisor.updater] Fetching update data from https://version.home-assistant.io/stable.json
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.check] Starting system checks with state CoreState.RUNNING
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.SECURITY/ContextType.CORE
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.FREE_SPACE/ContextType.SYSTEM
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.checks.base] Run check for IssueType.PWNED/ContextType.ADDON
21-03-15 07:57:39 INFO (MainThread) [supervisor.homeassistant.secrets] Request secret google_drive_backup_password
21-03-15 07:57:39 INFO (MainThread) [supervisor.host.services] Updating service information
21-03-15 07:57:39 INFO (MainThread) [supervisor.host.network] Updating local network information
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.module] Create new issue IssueType.PWNED - ContextType.ADDON / e92a3241_paradox_alarm_interface_dev
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.module] Create new suggestion SuggestionType.EXECUTE_STOP - ContextType.ADDON / e92a3241_paradox_alarm_interface_dev
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.check] System checks complete
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.fixup] Starting system autofix at state CoreState.RUNNING
21-03-15 07:57:39 INFO (MainThread) [supervisor.resolution.fixup] System autofix complete
21-03-15 07:57:40 INFO (MainThread) [supervisor.host.sound] Updating PulseAudio information
21-03-15 07:57:40 INFO (MainThread) [supervisor.host] Host information reload completed
21-03-15 08:04:55 INFO (MainThread) [supervisor.hardware.monitor] Detecting HardwareAction.REMOVE usb hardware /dev/bus/usb/001/003
21-03-15 08:05:11 INFO (MainThread) [supervisor.hardware.monitor] Detecting HardwareAction.ADD usb hardware /dev/bus/usb/001/004
21-03-15 08:05:21 INFO (MainThread) [supervisor.hardware.monitor] Detecting HardwareAction.REMOVE usb hardware /dev/bus/usb/001/004
21-03-15 08:05:51 INFO (MainThread) [supervisor.hardware.monitor] Detecting HardwareAction.ADD usb hardware /dev/bus/usb/001/005
21-03-15 08:16:18 INFO (MainThread) [supervisor.api.security] /network/info access from a0d7b954_ssh
21-03-15 08:16:18 INFO (MainThread) [supervisor.api.security] /host/info access from a0d7b954_ssh
21-03-15 08:16:18 INFO (MainThread) [supervisor.api.security] /core/info access from a0d7b954_ssh
21-03-15 08:16:39 INFO (MainThread) [supervisor.api.security] /supervisor/logs access from a0d7b954_ssh
```

**Journal logs:**

**Kernel logs:**

**Description of problem:**

From time to time the Bluetooth USB dongle stops working and I simply can't get it to work again.
This is the host log when the device is inserted:
```
[  559.500685] usb 1-2: USB disconnect, device number 4
[  587.116457] usb 1-2: new full-speed USB device number 5 using xhci_hcd
[  587.444249] usb 1-2: New USB device found, idVendor=0bda, idProduct=8771, bcdDevice= 2.00
[  587.444250] usb 1-2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  587.444251] usb 1-2: Product: Bluetooth Radio
[  587.444252] usb 1-2: Manufacturer: Realtek
[  587.444253] usb 1-2: SerialNumber: 00E04C239987
[  587.449150] Bluetooth: hci0: RTL: examining hci_ver=0a hci_rev=000b lmp_ver=0a lmp_subver=8761
[  587.450112] Bluetooth: hci0: RTL: rom_version status=0 version=1
[  587.450113] Bluetooth: hci0: RTL: loading rtl_bt/rtl8761a_fw.bin
[  587.450123] bluetooth hci0: Direct firmware load for rtl_bt/rtl8761a_fw.bin failed with error -2
[  587.450125] Bluetooth: hci0: RTL: firmware file rtl_bt/rtl8761a_fw.bin not found
```

This is not HassOS 5.12 related, it behaved the same in older versions also.
I cannot say how do I get it to work again, I simply try countless times to plug it back in or restart the VM and it will start to work at some point.

**Comment by agners** [comment]
This sounds like a missing firmware.

**Comment by clau-bucur** [comment]
It works most of the time, so shouldn't mean that the firmware is there?

**Comment by agners** [comment]
`-2` means "No such file or directory", and from what I can see Home Assistant OS has that firmware not installed. Not sure why it works sometime, some devices have a default firmware (which might be buggy and hence work only sometimes?), or it could be that your host Operating System's driver loads the firmware beforehand maybe?

**Comment by clau-bucur** [comment]
I see. Is there a way that I can add the firmware manually until it will be a standard part of HassOS ?

**Comment by kelos-01** [comment]
Not sure if I should log a bug/enhancement as well, seem to be having the same issue. I see the commit is completed but still in Dev. (btw thanks @agners)

```
[16263.319768] usb 1-1.4: USB disconnect, device number 4
[16642.754635] usb 1-1.2: new full-speed USB device number 5 using dwc_otg
[16642.887871] usb 1-1.2: New USB device found, idVendor=0bda, idProduct=8771, bcdDevice= 2.00
[16642.887899] usb 1-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[16642.887919] usb 1-1.2: Product: Bluetooth Radio
[16642.887936] usb 1-1.2: Manufacturer: Realtek
[16642.887953] usb 1-1.2: SerialNumber: 00E04C239987
[16642.907465] Bluetooth: hci0: RTL: examining hci_ver=0a hci_rev=000b lmp_ver=0a lmp_subver=8761
[16642.909456] Bluetooth: hci0: RTL: rom_version status=0 version=1
[16642.909486] Bluetooth: hci0: RTL: loading rtl_bt/rtl8761b_fw.bin
[16642.923271] bluetooth hci0: Direct firmware load for rtl_bt/rtl8761b_fw.bin failed with error -2
[16642.923306] Bluetooth: hci0: RTL: firmware file rtl_bt/rtl8761b_fw.bin not found
```

**Comment by clau-bucur** [comment]
@agners is this fix included in 5.13?
The 5.13 update does not show up in HA though.

**Comment by clau-bucur** [comment]
Looks like 5.13 does not contain the fix, as the behavior remains unchanged after 5.13 upgrade.

**Comment by GreyTeardrop** [comment]
I've upgraded to [OS 6.0.rc1](https://github.com/home-assistant/operating-system/releases/tag/6.0.rc1) that is supposed to have Realtek Bluetooth drivers bundled according to the release notes, but alas, I see the same issue. My host is ODroid N2+ (Home Assistant Blue)
```
~ $ bluetoothctl list
~ $ ha os info
board: odroid-n2
boot: B
update_available: false
version: 6.0.rc1
version_latest: 6.0.rc1
~ $ ha host logs|grep -i bluetooth
[    3.461272] Bluetooth: Core ver 2.22
[    3.465123] Bluetooth: HCI device and connection manager initialized
[    3.471016] Bluetooth: HCI socket layer initialized
[    3.475603] Bluetooth: L2CAP socket layer initialized
[    3.482178] Bluetooth: SCO socket layer initialized
[    3.497825] Bluetooth: hci0: RTL: examining hci_ver=0a hci_rev=000b lmp_ver=0a lmp_subver=8761
[    3.504819] Bluetooth: hci0: RTL: rom_version status=0 version=1
[    3.509350] Bluetooth: hci0: RTL: loading rtl_bt/rtl8761b_fw.bin
[    3.517721] bluetooth hci0: Direct firmware load for rtl_bt/rtl8761b_fw.bin failed with error -2
[    3.524450] bluetooth hci0: Falling back to sysfs fallback for: rtl_bt/rtl8761b_fw.bin
[   64.482850] Bluetooth: hci0: RTL: firmware file rtl_bt/rtl8761b_fw.bin not found
```

**Comment by wzaatar** [comment]
Confirming the issue is still there with OS 5.13 and Supervisor 2021.05.4 also on ODROID (HA Blue). 

I tried following the instructions at https://community.home-assistant.io/t/hass-io-and-usb-bluetooth-dongle/217917/4 but a reboot erases the folder and files.

The issue title should be changed to "fails all the time"...

**Comment by agners** [comment]
@clau-bucur @wzaatar the fix is in the development branch, so no changes in 5.13 or future 5 releases.

@GreyTeardrop @kelos-01 your devices seem to request a different firmware (note the `b` in `rtl_bt/rtl8761b_fw.bin`). I'll check if I can add this firmware as well.

**Comment by agners** [comment]
It seems that `rtl_bt/rtl8761b_fw.bin` is not available in the upstream linux-firmware repository. Splitting this firmware into a separate issue #1392.

The firmware @clau-bucur was mentioning (`rtl_bt/rtl8761a_fw.bin`) will be shipped in release 6 (see #1283). Hence closing.
